syntax = "proto3";

// option optimize_for = CODE_SIZE;
// option cc_enable_arenas = true;

package bftrader.bfcta;

import "bftrader.proto";

//
// Cta服务，必须先connect然后才可以调用其他api
//
service BfCtaService{    
    // 请求建立到robot的连接
    rpc Connect(BfConnectReq) returns (BfConnectResp) {}
    
    // 活跃检测
    rpc Ping(BfPingData) returns (BfPingData) {}
        
    // 请求断开到robot的连接
    rpc Disconnect(BfVoid) returns (BfVoid) {}
    
    // 获取策略信息，如position，用kv最方便
    rpc GetRobotInfo(BfKvData) returns (BfKvData) {}

    // 发单
    rpc SendOrder(BfSendOrderReq) returns (BfSendOrderResp) {}
    
    // 撤单
    rpc CancelOrder(BfCancelOrderReq) returns (BfVoid) {}
            
}